guide.md

```diff
@@ -0,0 +1,388 @@
+inserting a new record.
+
+```rust
+// src/models.rs
+use super::schema::posts;
+
+#[insertable_into(posts)]
+pub struct NewPost<'a> {
+    pub title: &'a str,
+    pub body: &'a str,
+}
+```
+
+Now let's add a function to save a new post.
+
+```rust
+// src/lib.rs
+use self::models::{Post, NewPost};
+
+pub fn create_post<'a>(conn: &PgConnection, title: &'a str, body: &'a str) -> Post {
+    use schema::posts;
+
+    let new_post = NewPost {
+        title: title,
+        body: body,
+    };
+
+    diesel::insert(&new_post).into(posts::table)
+        .get_result(conn)
+        .expect("Error saving new post")
+}
+```
+
+When we call `.get_result` on an insert or update statement, it automatically
+adds `RETURNING *` to the end of the query, and lets us load it into any struct
+that implements `Queryable` for the right types. Neat!
+
+#### Protip (This would be a sidebar or something)
+
+Diesel can insert more than one record in a single query. Just pass a `Vec` or
+slice to `insert`, and then call `get_results` instead of `get_result`. If you
+don't actually want to do anything with the row that was just inserted, call
+`.execute` instead. The compiler won't complain at you, that way. :)
+
+Now that we've got everything set up, we can create a little script to write a
+new post.
+
+```rust
+// src/bin/write_post.rs
+extern crate diesel_demo;
+extern crate diesel;
+
+use self::diesel_demo::*;
+use std::io::{stdin, Read};
+
+fn main() {
+    let connection = establish_connection();
+
+    println!("What would you like your title to be?");
+    let mut title = String::new();
+    stdin().read_line(&mut title).unwrap();
+    let title = &title[..(title.len() - 1)]; // Drop the newline character
+    println!("\nOk! Let's write {} (Press {} when finished)\n", title, EOF);
+    let mut body = String::new();
+    stdin().read_to_string(&mut body).unwrap();
+
+    let post = create_post(&connection, title, &body);
+    println!("\nSaved draft {} with id {}", title, post.id);
+}
+
+#[cfg(not(windows))]
+const EOF: &'static str = "CTRL+D";
+
+#[cfg(windows)]
+const EOF: &'static str = "CTRL+Z";
+```
+
+We can run our new script with `cargo run --bin write_post`. Go ahead and write
+a blog post. Get creative! Here was mine:
+
+```
+   Compiling diesel_demo v0.1.0 (file:///Users/sean/Documents/Projects/open-source/diesel_demo)
+     Running `target/debug/write_post`
+
+What would you like your title to be?
+Diesel demo
+
+Ok! Let's write Diesel demo (Press CTRL+D when finished)
+
+You know, a CLI application probably isn't the best interface for a blog demo.
+But really I just wanted a semi-simple example, where I could focus on Diesel.
+I didn't want to get bogged down in some web framework here.
+Plus I don't really like the Rust web frameworks out there. We might make a
+new one, soon.
+
+Saved draft Diesel demo with id 1
+```
+
+Unfortunately, running `show_posts` still won't display our new post, because we
```

Comment:
I think even here on this line. You could say "Unfortunately, running `show_posts` still won't display our new post because we published to default to false in the database earlier. We'll have to 'update' our post to publish it!" or something like that.